openmoc/materialize.py

```diff
@@ -355,3 +358,288 @@ def load_openmc_mgxs_lib(mgxs_lib, geometry=None):
 
     # Return collection of materials
     return materials
+
+
+##
+# @brief Compute SPH factors for an OpenMC multi-group cross section library
+# @details This routine coputes SuPerHomogenisation (SPH) factors for an
+#          an OpenMC MGXS library. The SPH scheme is outlined by Alain Hebert
+#          in the following paper:
+#
+#          Hebert, A., "A Consistent Technique for the Pin-by-Pin
+#          Homogenization of a Pressurized Water Reactor Assembly."
+#          Nuclear Science and Engineering, 113 (3), pp. 227-233, 1993.
+#
+#          The SPH factors are needed to preserve reaction rates in
+#          heterogeneous geometries. The energy condensation process leads
+#          to a bias between ultrafine and coarse energy group calculations.
+#          This bias is a result of the use of scalar flux-weighting to compute
+#          MGXS without properly accounting for angular-dependence of the flux.
+# @param mgxs_lib an openmc.mgxs.Library object
+# @param max_iters the maximum number of SPH iterations
+# @param sph_tol the tolerance on the SPH factor convergence
+# @param fix_src_tol the tolerance on the MOC fixed source calculations
+# @param num_azim number of azimuthal angles to use in fixed source calculations
+# @param track_spacing the track spacing to use in fixed source calculations
+# @return a NumPy array of SPH factors and a new openmc.mgxs.Library object
+#         with the SPH factors applied to each MGXS object
+def compute_sph_factors(mgxs_lib, max_iters=10, sph_tol=1E-6, fix_src_tol=1E-5,
+                        num_azim=4, track_spacing=0.1, zcoord=0.0):
+
+    import openmc.mgxs
```

Comment:
Can we have a try/catch around this so it can tell users about the dependency if they don't have openmc installed?